src/Stack/Build/ConstructPlan.hs

```diff
@@ -149,9 +149,13 @@ constructPlan mbp0 baseConfigOpts0 locals extraToBuild0 locallyRegistered loadPa
                 , planFinals = M.fromList finals
                 , planUnregisterLocal = mkUnregisterLocal tasks dirtyReason locallyRegistered
                 , planInstallExes =
-                    if boptsInstallExes $ bcoBuildOpts baseConfigOpts0
-                        then installExes
-                        else Map.empty
+                    let installExesPlan = 
+                            boptsInstallExes $ bcoBuildOpts baseConfigOpts0
+                        in case installExesPlan of
```

Comment:
This code still makes me uncomfortable, as you're encoding the same information as both a `Bool` and by making an empty `Map`. Is there any reason to keep the `Bool` around?